src/Core/Configurations/RuntimeConfigProvider.cs

```diff
@@ -160,6 +160,34 @@ public bool TryGetLoadedConfig([NotNullWhen(true)] out RuntimeConfig? runtimeCon
         return configLoadSucceeded;
     }
 
+    /// <summary>
+    /// Initialize the runtime configuration provider with the specified accessToken for the specified datasource.
+    /// This initialization method is used to set the access token for the current runtimeConfig.
+    /// As opposed to using a json input and regenerating the runtimconfig, it sets the access token for the current runtimeConfig on the provider.
+    /// </summary>
+    /// <param name="accessToken">The string representation of a managed identity access token</param>
+    /// <param name="dataSourceName">Name of the datasource for which to assign the token.</param>
+    /// <returns>true if the initialization succeeded, false otherwise.</returns>
+    public bool TryInitializeWithAccesstoken(
+        string? accessToken,
+        string dataSourceName)
+    {
+        if (_runtimeConfig is null)
+        {
+            // if runtimeConfig is not set up, throw as cannot initialize.
+            throw new DataApiBuilderException($"{nameof(RuntimeConfig)} has not been loaded.", HttpStatusCode.BadRequest, DataApiBuilderException.SubStatusCodes.ErrorInInitialization);
```

Comment:
Would the status code be BadRequest or something else? Essentially, can this exception occur at request time or initialization startup time? If latter, then BadRequest doesnt make sense.

Reply:
I was assuming this would be at request time, i.e,  you are explicitly calling this function to set up an existing runtimeconfig with the accesstoken after you have initialized the runtimeconfig already using appropriate loaders.